config/formats.js

```diff
@@ -297,12 +297,24 @@ exports.Formats = [
 		mod: 'gen7',
 		ruleset: ['[Gen 7] OU'],
 		banlist: ['Endeavor', 'Blast Burn + Explosion + Frenzy Plant + Giga Impact + Hydro Cannon + Hyper Beam + Self Destruct + V-Create > 2'],
-		onBeforeFaint: function (pokemon, source) {
-			try {
-				this.add('-hint', `${pokemon.name || pokemon.species}'s Last Will let it use one last move!`);
-				this.runMove(pokemon.moves[pokemon.moves.length - 1], pokemon);
-			} catch (e) {
-				this.add('-message', 'But it failed!');
+		onAfterMove: function (source, target) {
+			if (target.hp <= 0) {
+				try {
+					this.add('-hint', `${target.name || target.species}'s Last Will let it use one last move!`);
+					this.runMove(target.moves[target.moves.length - 1], target);
+				} catch (e) {
+					this.add('-message', 'But it failed!');
+				}
+			}
+		},
+		onAfterDamage: function (damage, target, source, effect) {
+			if (effect.effectType !== 'Move' && target.hp <= 0) {
```

Comment:
Looks like `BeforeFaint` is only used in Last Will. The simplest solution here is to move it to the actual fainting, rather than having it execute mid-move.